### ISSUE TITLE
Errors are printend in the output

### ISSUE THREAD

**Issue by roberth** [justinwoo/format-nix#4]
For example:

in:
```
pkgs: nixpkgs:

let
  foo = {}
in
{}
```

out:
```
pkgs: nixpkgs

Unknown ERROR :

let
  foo = {}
in
{}
```

**Comment by justinwoo** [comment]
this looks like some parsing failure, probably because the binding `foo = {}`

**Comment by roberth** [comment]
Correct, but the formatter should not produce files that are worse. If it can't parse, it should print a warning and leave the file as is. (I should've said that in the first place :) )

**Comment by justinwoo** [comment]
Maybe you then want to modify how expr2Doc works (https://github.com/justinwoo/format-nix/blob/db288d97a9018a50d3be63f8b20aa15e3dd45d0b/src/FormatNix.purs#L275) and handle that in Main https://github.com/justinwoo/format-nix/blob/db288d97a9018a50d3be63f8b20aa15e3dd45d0b/src/Main.purs#L29

**Comment by justinwoo** [comment]
probably you instead would want to just add error to the Expr data type and then have a validation function for Expr -> V NonEmptyList Expr (https://pursuit.purescript.org/packages/purescript-validation/4.2.0/docs/Data.Validation.Semigroup#t:V) but for a quick patch you could do the above

**Comment by gilligan** [comment]
Think i just ran into something similar..

**Input**
```nix
{ fetched ? import ./nixpkgs-fetch.nix {}
, nixpkgs ? fetched.pkgs
}:

let
  tools = import ./tools-overlay.nix;
  contrail = import ./contrail-overlay.nix;

in import nixpkgs { overlays = [ tools contrail ]; }

```

**Output**
```nix
{ fetched ? import ./nixpkgs-fetch.nix {}, nixpkgs ? fetched.pkgs }:

let
  Unknown binds Unknown binds tools = import ./tools-overlay.nix;
  contrail = import ./contrail-overlay.nix;

in import nixpkgs {
  overlays = [ tools contrail ];
}
```

**Comment by justinwoo** [comment]
I think there's some weird mixup of versions here, this works for me on master, but the version of tree-sitter-nix you have available is different from the corresponding format-nix code

**Comment by justinwoo** [comment]
## TODO

* If Expr contains Unknowns, print errors to stderr if possible and exit 1
* If Expr is clean, then the target file can be written and exit 0

**Comment by justinwoo** [comment]
Fixed in #21
